File: php/decorator/Salon.php
<?php
require_once dirname(__FILE__) . '/Decorators/ColoringDecorator.php';
require_once dirname(__FILE__) . '/Decorators/HaircutDecorator.php';
require_once dirname(__FILE__) . '/Decorators/StylingDecorator.php';
require_once dirname(__FILE__) . '/MaleSalonService.php';
require_once dirname(__FILE__) . '/FemaleSalonService.php';

class Salon
{
    function provideService()
    {
        $maleSalonService = new MaleSalonService();
        $maleSalonService = new HaircutDecorator($maleSalonService);
        $maleSalonService = new StylingDecorator($maleSalonService);
        $maleSalonService = new ColoringDecorator($maleSalonService);

        return $maleSalonService->getDescription();
    }
}

File: ruby/decorator/salon.rb
require_relative "Decorators/coloring_decorator"
require_relative "Decorators/styling_decorator"
require_relative "Decorators/haircut_decorator"
require_relative "female_salon_service"
require_relative "male_salon_service"

class Salon
  def provideService
    salonService = FemaleSalonService.new
    salonService = ColoringDecorator.new(salonService)
    salonService = StylingDecorator.new(salonService)
    salonService = HaircutDecorator.new(salonService)

    puts salonService.getDescription
  end
end